pydicom/tests/test_cli.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2020 pydicom authors. See LICENSE file for details.
+"""Tests for command-line interface"""
+
+from argparse import ArgumentTypeError
+
+import pytest
+
+from pydicom.cli.main import (
+    filespec_parser, eval_element, main, filespec_parts
+)
+
+
+bad_elem_specs = (
+    "extra:colon",
+    "no_callable()",
+    "no_equals = ",
+    "BeamSequence[0]extra",  # must match to end of string
+    "BeamSequence[x]",  # index must be an int
+)
+
+missing_elements = (
+    "NotThere",
+    "BeamSequenceXX",
+    "BeamDose",  # valid keyword but not at top level
+)
+
+bad_indexes = (
+    "BeamSequence[42]",
+    "BeamSequence[-42]",
+)
+
+
+class TestFilespec:
+    @pytest.mark.parametrize("bad_spec", bad_elem_specs)
+    def test_syntax(self, bad_spec):
+        """Invalid syntax for for CLI file:element spec raises error"""
+        with pytest.raises(ArgumentTypeError, match=r".* syntax .*"):
+            filespec_parser(f"pydicom::rtplan.dcm::{bad_spec}")
+
+    @pytest.mark.parametrize("missing_element", missing_elements)
+    def test_elem_not_exists(self, missing_element):
+        """CLI filespec elements not in the dataset raise an error"""
+        with pytest.raises(
+            ArgumentTypeError, match=r".* is not in the dataset"
+        ):
+            filespec_parser(f"pydicom::rtplan.dcm::{missing_element}")
+
+    @pytest.mark.parametrize("bad_index", bad_indexes)
+    def test_bad_index(self, bad_index):
+        """CLI filespec elements with an invalid index raise an error"""
+        with pytest.raises(ArgumentTypeError, match=r".* index error"):
+            filespec_parser(f"pydicom::rtplan.dcm::{bad_index}")
+
+    def test_offers_pydicom_testfile(self):
+        """CLI message offers pydicom data file if file not found"""
+        with pytest.raises(
+            ArgumentTypeError, match=r".*pydicom::rtplan\.dcm.*is available.*"
+        ):
+            filespec_parser(f"rtplan.dcm")
+
+    def test_colons(self):
+        """CLI filespec with a colon in filename works correctly"""
+        expected = ("", r"c:\test.dcm", "")
+        assert expected == filespec_parts(r"c:\test.dcm")
+
+        expected = ("pydicom", r"c:\test.dcm", "")
+        assert expected == filespec_parts(r"pydicom::c:\test.dcm")
+
+        filespec = r"pydicom::c:\test.dcm::StudyDate"
+        expected = ("pydicom", r"c:\test.dcm", "StudyDate")
+        assert expected == filespec_parts(filespec)
+
+        filespec = r"c:\test.dcm::StudyDate"
+        expected = ("", r"c:\test.dcm", "StudyDate")
+        assert expected == filespec_parts(filespec)
+
+
+class TestFilespecElementEval:
+    # Load plan once
+    plan, _ = filespec_parser("pydicom::rtplan.dcm")[0]
```

Comment:
could move that into the test - no need for a class variable here (but also no harm...)

Reply:
Good point - I probably started with some of the tests in separate methods, or thought I was going to add more.  Think I'll leave there for possible future expansion.